crates/sovereign-sdk/rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -48,12 +48,26 @@ pub trait ZkvmHost: Zkvm + Clone {
     /// This runs the guest binary compiled for the zkVM target, optionally
     /// creating a SNARK of correct execution. Running the true guest binary comes
     /// with some mild performance overhead and is not as easy to debug as [`simulate_with_hints`](ZkvmHost::simulate_with_hints).
-    fn run(&mut self, with_proof: bool) -> Result<Proof, anyhow::Error>;
+    fn run(&mut self, with_proof: bool, l1_block_height: u64) -> Result<Proof, anyhow::Error>;
 
     /// Extracts public input form the proof.
     fn extract_output<Da: DaSpec, Root: BorshDeserialize>(
         proof: &Proof,
     ) -> Result<StateTransition<Da, Root>, Self::Error>;
+
+    /// Waits for a receipt to be generated by the zkVM program.
+    fn wait_for_receipt(&self, session: &str) -> Result<Vec<u8>, anyhow::Error>;
+
+    /// Creates a new snark session in the zkVM program.
+    fn create_new_snark_session(&self, session: &str) -> Result<String, anyhow::Error>;
```

Comment:
Just realized that this won't work in no-std because of String 
I will update it with a serialized version

Reply:
Fixed